### PR TITLE
Pull in various changes from fork

### PR DESCRIPTION
**Pull request by Neo-Zhixing**
Now that @gwihlidal has given maintainer permissions to me, there is no longer a need to maintain a fork of this repository.
This PR pulls in various changes from the fork.
- Add setter for VirtualBlockInfo::flag
- Indicate that VirtualBlock is not thread safe
- destroy_image to take an Allocation ref
- Mark get_allocation_info as safe
- Fix tests

### CODE REVIEW

File: Cargo.toml
Comment (MarijnS95) [inline Cargo.toml:1]: Did line-endings change in this file and `wrapper.cpp`?

Comment (Neo-Zhixing) [reply Cargo.toml:1]: No I just moved some files around.

Comment (MarijnS95) [reply Cargo.toml:1]: This file didn't move, and the line-ending change appears to be gone :+1:

Comment (Neo-Zhixing) [reply Cargo.toml:1]: Not exactly sure what was going on there 😕

**Pull request merged**